recipes/kubernetes-helm/build.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+make_goroot_read_only()
+{
+    find $GOROOT -type d -exec chmod 555 {} \;
+}
+
+make_gopath()
+{
+   export GOPATH=$(pwd)/go
+   export PATH=$GOPATH/bin:$PATH
+
+   export GOSRC_DIR=$GOPATH/src/k8s.io/helm
+   mkdir -p $GOSRC_DIR
+
+   find . -maxdepth 1 \
+       -not -name '.' -and \
+       -not -name 'go' -and \
+       -not -name 'LICENSE' \
+       -exec mv {} $GOSRC_DIR/ \;
+   cd $GOSRC_DIR
```

Comment:
Could you please explain this a bit for me? Looks like we are moving everything into a source directory for building, but am not following all the exclusions or the reason for the location. It may just be that you are way more knowledgeable about best practices in go. Sorry if this is an obvious thing.

Reply:
GO usually wants all the source code to be in under  $GOPATH/src/<fully-qualified-path>, where fully-qualified-path changes from package to package, e.g. for kubernetes-helm it is k8s.io/helm, for terraform it is github.com/hashicorp/terraform, etc. 

During build, it is common for the build scripts to download build-time dependencies to $GOPATH/src, compile and place them under $GOPATH/bin. This is why we don't want $GOPATH to coincide with $PREFIX, o.w. we may end up packaging build dependencies with the final package. 

So... the hack/workaround/solution is to create a GOPATH folder outside of PREFIX and move all the code into its fully-qualified-path. I happen to choose "./go" as the GOPATH folder. Then, the find exclusions make sure that the GOPATH does not move inside itself and that the LICENSE file stays at its default location so meta.yaml looks like the canonical one without the fully-qualified-path.

Reply:
Makes sense. Maybe the `GOPATH` hack can be added to `go` as part of the activation script and only run during a `conda-build` run.

Reply:
I think this might make more sense in the toolchain package, what do you think?

Reply:
So we will be moving away from the `toolchain` package when we switch to the new Anaconda compilers.

FWIW [we do a similar thing with `setuptools`]( https://github.com/conda-forge/setuptools-feedstock/blob/2ba01738abf79f1f37a2c72480aa65f9f2d064f0/recipe/nodownload.patch ). Hence the suggestion to add it to `go`.